ldms/src/core/ldms.c

```diff
@@ -1032,11 +1032,13 @@ size_t __ldms_value_size_get(enum ldms_value_type t, uint32_t count)
 	return roundup(value_sz, 8);
 }
 
-void __ldms_metric_size_get(const char *name, enum ldms_value_type t,
-		uint32_t count, size_t *meta_sz, size_t *data_sz)
+void __ldms_metric_size_get(const char *name, const char *unit,
+			    enum ldms_value_type t,
+			    uint32_t count, size_t *meta_sz, size_t *data_sz)
 {
 	/* Descriptors are aligned on eight byte boundaries */
```

Comment:
ldms_value_desc is preceded by "pragma pack(4)". 4 means 4 bytes, right? Why are we assuming 8 byte/ 64-bit boundaries? Isn't that a bug?

Reply:
We're not assuming, we're requiring 4B alignment. These structures are RDMA'd around, so the consumer and producer have to have the exact same layout.

Reply:
Right, 4B. But the code says 8B. Why is that?

Reply:
@morrone The structure itself if 4B pack. However, the structure has a variable length data. The start/end of the entire structure+data has to be 8B aligned due to RDMA requirement.